pytorch_lightning/plugins/training_type/parallel.py

```diff
@@ -34,10 +36,17 @@ def __init__(
     ):
         super().__init__()
         self.parallel_devices = parallel_devices
-        self.local_rank = 0
         self.world_size = 1
+        self.local_rank = 0
         self.cluster_environment = cluster_environment
 
+    @property
+    def cluster_local_rank(self):
+        try:
+            return self.cluster_environment.local_rank()
+        except KeyError:
+            return 0
+
```

Comment:
What is this for?

Reply:
cc @tchaton 
currently used in ddp_spawn.py line 94
I plan to revisit the cluster environments here #5915 to improve the overall integration, usability and documentation.

Reply:
Yes, definitely. It is a temporary solution until we move `setup_optimizers`, `setup precision_plugin.setup()` within spawn. Great catch though !

Reply:
It was used to assign `task_idx` to the training_type for the test.
And `self.cluster_environment.local_rank()` should fail if the "LOCAL_RANK" isn't in os.environ.